tools/codegen/api/types.py

```diff
@@ -26,13 +26,58 @@
 # you need trnsnlations that know about these types, beef up this data
 # structure.
 
-@dataclass(frozen=True)
+
+# The set of all non-templated, valid, fully-qualified names of C++ types that are used in the codegen.
+# Templated types get their own dataclass, mainly to make namespace parsing easier.
+BaseCppType = Enum('BaseCppType', (
+    'int64_t',
+    'double',
+    'bool',
+    'void',
+    'std::string',
+    'at::Generator',
```

Comment:
One thing that's a little goofy is that for half of these types, `at::` isn't even the canonical namespace; they were originally defined in `c10::`. It's probably OK if the codegen studiously pretends that c10:: namespace doesn't exist though

Reply:
Byte for byte isn't a fundamentalist religion: you can violate it if it makes sense :) (Just more debugging out the other end!)

Reply:
yeah :p I've found it to be really helpful in uncovering diffs between my codegen output's current state -> target state. I could always just keep it byte-for-byte locally though until I'm satisfied, then remove pointless diffs before I merge.